tests/components/librehardwaremonitor/test_sensor.py

```diff
@@ -0,0 +1,132 @@
+"""Test the LibreHardwareMonitor sensor."""
+
+import copy
+from datetime import timedelta
+from unittest.mock import AsyncMock
+
+from freezegun.api import FrozenDateTimeFactory
+from librehardwaremonitor_api import LibreHardwareMonitorConnectionError
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.librehardwaremonitor.const import DEFAULT_SCAN_INTERVAL
+from homeassistant.const import STATE_UNAVAILABLE, STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_sensors_are_created(
+    hass: HomeAssistant,
+    mock_lhm_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    freezer: FrozenDateTimeFactory,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test sensors are created."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    initial_states = hass.states.async_all()
+    assert initial_states == snapshot(name="valid_sensor_data")
+
+    mock_lhm_client.get_data.side_effect = LibreHardwareMonitorConnectionError()
+
+    freezer.tick(timedelta(DEFAULT_SCAN_INTERVAL))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    unavailable_states = hass.states.async_all()
+    assert all(state.state == STATE_UNAVAILABLE for state in unavailable_states)
+
+    mock_lhm_client.get_data.side_effect = None
+
+    freezer.tick(timedelta(DEFAULT_SCAN_INTERVAL))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    recovered_states = hass.states.async_all()
+    assert recovered_states == snapshot(name="valid_sensor_data")
```

Comment:
Snapshots are cool. They help with fixating data and tracking change. What they are bad at, is tracking change in the same test. I would rather see a harder assertion here, so we don't hide these details in the ambr file